### DISCUSSION TITLE
/v2/chart/balance/balance ignoring the `period` parameter

### DISCUSSION THREAD

**Discussion by victorbalssa** [firefly-iii/firefly-iii#9271] [General]
Following https://api-docs.firefly-iii.org/?urls.primaryName=2.0.14%20(v2)#/charts/getBalanceChart endpoint `/v2/chart/balance/balance` should return subset of `entries` based on `period` value (e.g. `1D`, `1W`, `1M`, `3M`, `6M`, `1Y`).

It always return only one entry and ignore the `period` filter.

Is it a bug or should I use another endpoint? @JC5 

Thanks again!

(https://github.com/victorbalssa/abacus/issues/331#issuecomment-2366860086)

**Comment by JC5** [comment]
I'll check it out, thanks!

**Comment by JC5** [comment]
This is currently impossible to document, because Swagger doesn't support deep array objects, but this is a valid format, and should work as expected:

```
https://{{host}}/api/v2/chart/balance/balance?filter[start]=2024-09-01&filter[end]=2024-09-30&filter[period]=1D&filter[accounts][]=1&filter[accounts][]=2&filter[preselected]=all
```

preselected can be "all", "assets" or "liabilities" and will overrule any accounts in the "accounts" array.

**Comment by victorbalssa** [comment]
It works thanks! (deep array objects)

`filter[period]=1D` isn't valid for this endpoint tho.

**Reply by JC5** [reply]
It should be, but no matter. This endpoint will change in the near future because deep arrays are impossible to document in Swagger. I'll keep u posted of course.
